saas/models/saas_template.py

```diff
@@ -0,0 +1,144 @@
+# Copyright 2018 Ivan Yelizariev <https://it-projects.info/team/yelizariev>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+import random
+import string
+import logging
+
+from odoo import models, fields, api
+from odoo.tools.safe_eval import test_python_expr, safe_eval
+from odoo.exceptions import ValidationError
+from odoo.addons.queue_job.job import job
+from ..xmlrpc import rpc_auth, rpc_install_modules, rpc_code_eval
+
+_logger = logging.getLogger(__name__)
+
+MANDATORY_MODULES = ['auth_quick']
+
+
+def random_password(len=32):
+    return ''.join(random.SystemRandom().choice(string.ascii_uppercase + string.digits) for _ in range(len))
+
+
+class SAASTemplate(models.Model):
+    _name = 'saas.template'
+    _description = 'Database Template'
+
+    name = fields.Char()
+    template_demo = fields.Boolean('Install demo data', default=False)
+    template_modules_domain = fields.Text(
+        'Modules to install',
+        help='Domain to search for modules to install after template database creation',
+        defalt="[]")
+    template_post_init = fields.Text(
+        'Template Initialization',
+        default=lambda s: s.env['ir.actions.server'].DEFAULT_PYTHON_CODE,
+        help='Python code to be executed once db is created and modules are installed')
+    operator_ids = fields.One2many(
+        'saas.template.operator',
+        'template_id')
+
+    @api.constrains('code')
```

Comment:
@yelizariev, shouldn't there be template_modules_domain field here?

Reply:
Right. It's supposed to be ``template_post_init`` here :+1: